Util/Xamarin.Build.Download/source/Xamarin.Build.Download.Tests/Test.cs

```diff
@@ -448,9 +448,90 @@ public void testAndroidAarAdded (bool useAndroidSdk)
 		}
 
 		[Test]
-		public void TestGetItemsToDownload ()
+		public void TestDisallowUnsafeGetItemsToDownload ()
+		{
+			var itemUrl = "http://search.maven.org/remotecontent?filepath=com/facebook/android/facebook-android-sdk/4.17.0/facebook-android-sdk-4.17.0.aar";
+			var zipUrl = "http://dl-ssl.google.com/android/repository/android_m2repository_r40.zip";
+
+			var engine = new ProjectCollection ();
+			var prel = ProjectRootElement.Create (Path.Combine (TempDir, "project.csproj"), engine);
+
+			var unpackDir = GetTempPath ("unpacked");
+			prel.SetProperty ("XamarinBuildDownloadDir", unpackDir);
+
+			prel.AddItem (
+				"XamarinBuildDownload", "FacebookAndroid-4.17.0", new Dictionary<string, string> {
+					{ "Url", itemUrl },
+					{ "Kind", "Uncompressed" },
+				});
+
+			prel.AddItem (
+				"XamarinBuildDownloadPartialZip", "androidsupport-25.0.1/cardview.v7", new Dictionary<string, string> {
+					{ "Url", zipUrl },
+					{ "ToFile", "cardview.v7.aar" },
+					{ "RangeStart", "196438127" },
+					{ "RangeEnd", "196460160" },
+					{ "Md5", "b44eb88f7cc621ae616744c6646f5b64" }
+				});
+
+			AddCoreTargets (prel);
+
+			var project = new ProjectInstance (prel);
+			var log = new MSBuildTestLogger ();
+
+			var success = BuildProject (engine, project, "XamarinBuildDownloadGetItemsToDownload", log);
+
+			var itemToDownload = project.GetItems ("XamarinBuildDownloadItemToDownload");
+
+			Assert.IsEmpty (itemToDownload);
```

Comment:
I am not sure if it is just me, but my build does not fail when there is an error logged.